### PR TITLE
Ctcp 2831

### PR DESCRIPTION
**Pull request by srinivasulumunagalahmrc**

### CODE REVIEW

File: app/models/referenceData/CustomsOffice.scala [hunk 1]
```diff
@@ -18,7 +18,10 @@ package models.referenceData
 
 import play.api.libs.json.{Json, OFormat}
 
-case class CustomsOffice(id: String, name: String, roles: Seq[String], phoneNumber: Option[String])
+case class CustomsOffice(id: String, name: String, phoneNumber: Option[String]) {
+  val nameOption  = if (name.isEmpty) None else Some(name)
+  val phoneOption = if (phoneNumber.isDefined && !phoneNumber.get.isEmpty) Some(phoneNumber.get) else None
+}
```
Comment (MatthewAlexanderBeck) [inline app/models/referenceData/CustomsOffice.scala:24]: phoneNumber.filter(_.nonEmpty)?

Returns this Option if it is nonempty and applying the predicate p to this Option's value returns true. Otherwise, return None.
This is equivalent to:
 option match {
   case Some(x) if p(x) => Some(x)
   case _               => None
 }
 
Params:
p – the predicate used for testing.

File: app/viewModels/P5/departure/GoodsUnderControlP5ViewModel.scala [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    def fetchWhatHappensNext(ie060MessageData: IE060MessageData, customsOffice: Option[CustomsOffice])(implicit messages: Messages): String =
+      customsOffice match {
+        case Some(CustomsOffice(_, name, Some(phone))) =>
+          (name.nonEmpty, phone.nonEmpty) match {
+            case (true, true) => messages("goodsUnderControl.telephoneAvailable", name, phone)
+            case _            => messages("goodsUnderControl.telephoneNotAvailable", name)
+          }
+        case Some(CustomsOffice(_, name, None)) if name.nonEmpty        => messages("goodsUnderControl.telephoneNotAvailable", name)
+        case Some(CustomsOffice(id, "", Some(phone))) if phone.nonEmpty => messages("goodsUnderControl.teleAvailAndOfficeNameNotAvail", id, phone)
+        case _ =>
```
Comment (MatthewAlexanderBeck) [inline app/viewModels/P5/departure/GoodsUnderControlP5ViewModel.scala:52]: This looks repeated in the CustomsOfficeContactViewModel, can this be simplified?

File: conf/application.conf [hunk 3]
```diff
@@ -127,15 +127,22 @@ microservice {
...
     # otherwise users in production will be pointed to new phase 5 frontends
     features {
       isGuaranteeBalanceEnabled = true
       phase5Enabled {
-        departure = false
+        departure = true
         unloading = false
         cancellation = false
-        arrival = false
+        arrival = true
```
Comment (MatthewAlexanderBeck) [inline conf/application.conf:145]: Toggle back to false please

File: conf/testOnlyDoNotUseInAppConf.routes [hunk 4]
```diff
@@ -47,6 +47,11 @@ GET         /manage-transit-movements/test-only/view-arrival-notifications
 + nocsrf
 GET         /manage-transit-movements/test-only/view-departure-declarations                 controllers.testOnly.ViewAllDeparturesP5Controller.onPageLoad()
 
+
++ nocsfr
+GET         /manage-transit-movements/test-only/goods-under-control/:departureId   controllers.testOnly.TestOnlyGoodsUnderControlP5Controller.onPageLoad(departureId: String)
```
Comment (MatthewAlexanderBeck) [inline conf/testOnlyDoNotUseInAppConf.routes:52]: Just released but I dont think we need the 'nocsfr' bit for pages. This is only needed for calls to the API. Might be worth trying to remove this.

File: test/base/AppWithDefaultMockFixtures.scala [hunk 5]
```diff
@@ -37,4 +42,13 @@ trait AppWithDefaultMockFixtures extends GuiceOneAppPerSuite with BeforeAndAfter
       .overrides(
         bind[IdentifierAction].to[FakeIdentifierAction]
       )
+
+  protected def goodsUnderControlAction(departureIdP5: String,
+                                        mockDepartureP5MessageService: DepartureP5MessageService,
+                                        mockReferenceDataService: ReferenceDataService
+  ): Unit =
+    when(mockGoodsUnderControlActionProvider.apply(any())) thenReturn new FakeGoodsUnderControlAction(departureIdP5,
```
Comment (MatthewAlexanderBeck) [inline test/base/AppWithDefaultMockFixtures.scala:50]: Could be possibly do this at the controller level? Keep in mind that this AppWithDefaultFixtures is also used on the P4 pages. Makes sense to keep it locally to the controller we are using rather than globally.

File: test/connectors/DepartureMovementP5ConnectorSpec.scala [hunk 6]
```diff
@@ -45,48 +46,48 @@ class DepartureMovementP5ConnectorSpec extends SpecBase with WireMockServerHandl
...
+                "id": "6365135ba5e821ee",
+                "movementReferenceNumber": "27WF9X1FQ9RCKN0TM3",
+                "created": "2022-11-04T13:27:55.522Z",
+                "updated": "2022-11-04T13:27:55.522Z",
+                "enrollmentEORINumber": "9999912345",
+                "movementEORINumber": "GB1234567890"
+              }
+            ]
+          }
+          """.stripMargin
```
Comment (MatthewAlexanderBeck) [inline test/connectors/DepartureMovementP5ConnectorSpec.scala:90]: Dont need stripMargine is the '|' has been removed

File: conf/application.conf [hunk 7]
```diff
@@ -127,15 +127,22 @@ microservice {
...
       phase5Enabled {
-        departure = false
-        unloading = false
-        cancellation = false
-        arrival = false
+        departure = true
+        unloading = true
+        cancellation = true
+        arrival = true
       }
```
Comment (MatthewAlexanderBeck) [inline conf/application.conf:146]: Set these to false please (see comment above)

File: test/viewModels/CustomsOfficeContactViewModelSpec.scala [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+    "must render correct paragraph" - {
+
+      "When Customs office name and telephone exists" in {
+
+        val customsOffice     = CustomsOffice("ID001", "Dover", Some("00443243543"))
+        val viewModelProvider = CustomsOfficeContactViewModel("GB000060", Some(customsOffice))
+
+        val result: String = viewModelProvider.fetchWhatHappensNext
+        println(result)
```
Comment (MatthewAlexanderBeck) [inline test/viewModels/CustomsOfficeContactViewModelSpec.scala:37]: Can this be removed?

File: app/connectors/ReferenceDataConnector.scala [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  def getControlTypes()(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Option[Seq[ControlType]]] = {
+    val serviceUrl = s"${config.referenceDataUrl}/control-types"
+    http
+      .GET[Seq[ControlType]](serviceUrl)
+      .map(Some(_))
+      .recover {
+        case _ =>
+          println(s"$serviceUrl")
```
Comment (MatthewAlexanderBeck) [inline app/connectors/ReferenceDataConnector.scala:48]: And this please

**Review by jackheslop96 — APPROVED** [review]

**Review by ajchilts — DISMISSED** [review]

File: app/viewModels/P5/departure/GoodsUnderControlP5ViewModel.scala [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+
+  class GoodsUnderControlP5ViewModelProvider @Inject() (referenceDataService: ReferenceDataService) {
+
+    def apply(
+      ie060MessageData: IE060MessageData
+    )(implicit messages: Messages, ec: ExecutionContext, hc: HeaderCarrier): Future[GoodsUnderControlP5ViewModel] = {
+      val helper = new GoodsUnderControlP5MessageHelper(ie060MessageData, referenceDataService)
+
+      val notificationType: String    = ie060MessageData.TransitOperation.notificationType
+      val requestedDocuments: Boolean = ie060MessageData.requestedDocumentsToSeq.nonEmpty || notificationType == "1"
```
Comment (jackheslop96) [inline app/viewModels/P5/departure/GoodsUnderControlP5ViewModel.scala:80]: Could maybe look to introduce a NotificationType trait to avoid hardcoded "1" strings everywhere. Happy for this to be in a separate PR.

**Review by jackheslop96 — DISMISSED** [review]

**Review by kenneth-rayner — DISMISSED** [review]

**Review by kenneth-rayner — APPROVED** [review]
